### PR TITLE
Create backing User objects

### PR DESCRIPTION
**Pull request by wetherc**
Refs #75

This adds Pothos and Objection User objects as a precursor
to a user account creation flow. This also adds a utility
using argon2 to salt and pepper user passwords for storage
in the database.

### CODE REVIEW

File: dstk-infra/apollo/src/graphql/model-version/modelVersion.ts [hunk 1]
```diff
@@ -3,35 +3,37 @@ import { MLModel, ObjectionMLModel } from '../model/model.js';
...
 
-export const MLModelVersion = builder.objectRef<ObjectionMLModelVersion>('MLModelVersion').implement({
-    fields: (t) => ({
-        modelVersionId: t.field({
-            type: 'ID',
-            resolve(root: ObjectionMLModelVersion, _args, _ctx) {
-                return root.$modelClass.idColumn[0];
-            },
-        }),
+export const MLModelVersion = builder
```
Comment (wetherc) [inline dstk-infra/apollo/src/graphql/model-version/modelVersion.ts:6]: IDK why husky decided to pick up these lints now but 🤷

Comment (StephenODea54) [reply dstk-infra/apollo/src/graphql/model-version/modelVersion.ts:6]: What was the eslint complaining about?

**Review by StephenODea54 — APPROVED** [review]
LGTM. This stuff is gonna be fun lolol.

File: dstk-infra/apollo/src/graphql/model-version/modelVersion.ts (hunk 1 above)
Comment (wetherc) [reply dstk-infra/apollo/src/graphql/model-version/modelVersion.ts:6]: I didn't even notice it, the recommit hook just auto formatted this when I made the commit

**Pull request merged**